extensions/amp-subscriptions/0.1/entitlement-store.js

```diff
@@ -73,12 +73,29 @@ export class EntitlementStore {
     }
 
     this.firstResolvedPromise_ = new Promise(resolve => {
-      this.onChange(({entitlements}) => {
+      const entitlementsResolved = Object.keys(this.entitlements_).length;
+
+      // Check if current entitlements unblocks the reader
+      for (const key in this.entitlements_) {
+        const entitlements = (this.entitlements_[key]);
         if (entitlements.enablesThis()) {
-          resolve(entitlements);
+          return resolve(entitlements);
         }
-      });
+      }
+
+      if (entitlementsResolved === this.serviceIds_.length) {
+        // Resolve with null if non of the entitlements unblocks the reader
+        return resolve(null);
+      } else {
+        // Listen if any upcoming entitlements unblock the reader
+        this.onChange(({entitlements}) => {
+          if (entitlements.enablesThis()) {
+            return resolve(entitlements);
```

Comment:
What does this `return` mean? Overall, `return` in constructors are very dangerous.

Reply:
yeah this return isn't really necessary, although the ones above are just to make sure we are dont execute once we resolve